src/app.rs

```diff
@@ -973,12 +1002,46 @@ impl App {
     }
   }
 
-  pub fn user_follow_show(&mut self) {
-    unimplemented!();
+  pub fn user_follow_show(&mut self, block: ActiveBlock) {
+    match block {
+      ActiveBlock::SearchResultBlock => {
+        if let Some(shows) = &self.search_results.shows {
+          if let Some(selected_index) = self.search_results.selected_shows_index {
+            let show_id = shows.items[selected_index].id.to_owned();
```

Comment:
This could use `get` in case the index is somehow out of range?

```suggestion
            if let Some(show_id) = shows.items.get(selected_index).map(|item| item.id) {
               ...
            };
```